packages/flutter_availability/lib/src/routes.dart

```diff
@@ -18,23 +19,31 @@ MaterialPageRoute homePageRoute(VoidCallback onExit) => MaterialPageRoute(
     );
 
 ///
-MaterialPageRoute<AvailabilityTemplateModel?> templateOverviewRoute() =>
+MaterialPageRoute<AvailabilityTemplateModel?> templateOverviewRoute({
+  bool allowSelection = false,
+}) =>
     MaterialPageRoute(
       builder: (context) => AvailabilityTemplateOverview(
         onExit: () => Navigator.of(context).pop(),
         onEditTemplate: (template) async {
           if (template.templateType == AvailabilityTemplateType.day) {
             await Navigator.of(context).push(templateEditDayRoute(template));
+          } else if (template.templateType == AvailabilityTemplateType.week) {
+            await Navigator.of(context).push(templateEditWeekRoute(template));
           }
         },
         onAddTemplate: (type) async {
           if (type == AvailabilityTemplateType.day) {
             await Navigator.of(context).push(templateEditDayRoute(null));
+          } else if (type == AvailabilityTemplateType.week) {
+            await Navigator.of(context).push(templateEditWeekRoute(null));
           }
         },
-        onSelectTemplate: (template) async {
-          Navigator.of(context).pop(template);
-        },
+        onSelectTemplate: allowSelection
```

Comment:
I really don't like ternary operators when defining functions. There should be a way to do this in a more elegant manner